File: packages/emitter/testcases/common/expected/types/expected-type-threading/TernaryTyping.cs
namespace TestCases.common.types.expectedtypethreading
{
    [global::Tsonic.Internal.ModuleContainerAttribute]
    public static class TernaryTyping
    {
        public static readonly int ternaryInt = true ? (int)5 : (int)10;

        public static readonly int nestedTernary = true ? false ? (int)1 : (int)2 : (int)3;

        public static int ternaryReturn(bool flag)
        {
            return flag ? (int)100 : (int)200;
        }
    }
}

File: packages/emitter/testcases/common/types/expected-type-threading/TernaryTyping.ts
import { int } from "@tsonic/core/types.js";

// Ternary with int type annotation - both branches should be int
export const ternaryInt: int = true ? 5 : 10;

// Nested ternary - all branches should be int
export const nestedTernary: int = true ? (false ? 1 : 2) : 3;

// Function returning int via ternary
export function ternaryReturn(flag: boolean): int {
  return flag ? 100 : 200;
}
